org.lflang/src/org/lflang/generator/LFGenerator.java

```diff
@@ -54,50 +57,60 @@ public class LFGenerator extends AbstractGenerator {
      * @return A FileConfig object in Kotlin if the class can be found.
      * @throws IOException If the file config could not be created properly
      */
-    private FileConfig createFileConfig(final Target target,
-                                        Resource resource,
-                                        IFileSystemAccess2 fsa,
-                                        LFGeneratorContext context) throws IOException {
-        Path srcGenBasePath = FileConfig.getSrcGenRoot(fsa);
+    public static FileConfig createFileConfig(Resource resource, Path srcGenBasePath,
+                                        boolean useHierarchicalBin) {
+
+        final Target target = Target.fromDecl(ASTUtils.targetDecl(resource));
+        assert target != null;
+
         // Since our Eclipse Plugin uses code injection via guice, we need to
         // play a few tricks here so that FileConfig does not appear as an
-        // import. Instead we look the class up at runtime and instantiate it if
+        // import. Instead, we look the class up at runtime and instantiate it if
         // found.
-        switch (target) {
-        case CPP:
-        case Rust:
-        case TS:
-            String className = "org.lflang.generator." + target.packageName + "." + target.classNamePrefix + "FileConfig";
-            try {
-                return (FileConfig) Class.forName(className)
-                                         .getDeclaredConstructor(Resource.class, Path.class, boolean.class)
-                                         .newInstance(resource, srcGenBasePath, context.useHierarchicalBin());
-            } catch (InvocationTargetException e) {
-                throw new RuntimeException("Exception instantiating " + className, e.getTargetException());
-            } catch (ReflectiveOperationException e) {
-                return new FileConfig(resource, srcGenBasePath, context.useHierarchicalBin());
+        try {
+            if (FedASTUtils.findFederatedReactor(resource) != null) {
+                return new FedFileConfig(resource, srcGenBasePath, useHierarchicalBin);
+            }
+            switch (target) {
+            case CCPP:
+            case C: return new CFileConfig(resource, srcGenBasePath, useHierarchicalBin);
+            case Python: return new PyFileConfig(resource, srcGenBasePath, useHierarchicalBin);
+            case CPP:
+            case Rust:
+            case TS:
```

Comment:
It seems like we are still instantiating the target specific file configs here. So why are there all the changes to the target file configs?